Default the generate key frame algorithm to all layers.

jan-ivar opened the pull request:
From https://github.com/w3c/webrtc-encoded-transform/issues/143#issuecomment-1261963837

fippo commented:
What is the purpose of blocking the return until all requested key frames have been generated? This was previously required done to get "a" timestamp but no longer serves any purpose?

See also #145 on the overall broken assumptions

henbos commented:
Note that RIDs are optional, we only know that either all encodings have RIDs or none of them have RIDs, but we don't know that no RIDs equals singlecast - you could have multiple encodings without RIDs. The alternative would be encodingIndex but we can say that rids are a prerequisite

jan-ivar commented:
Note I also cleaned up the language a bit:

_"encoder"_ was being used at two different conceptual levels: A) as the owner of all layers, and B) per layer. I reworked/replaced the latter with _"layer"_ consistent with existing text in [4.10](https://w3c.github.io/webrtc-encoded-transform/#RTCRtpScriptTransformer-events): _"When transform’s [[encoder]] receives a keyframe request, ... Set rid to the RID of the appropriate layer, or undefined if the request is not for a specific layer."_

jan-ivar commented on index.bs:
```diff
@@ -1003,32 +1004,35 @@ The <dfn abstract-op>generate key frame algorithm</dfn>, given |promise|, |encod
+    1. Let |layers| be a new [=list=] of the layers for this |encoder|, ordered by negotiated encoding index.
+    1. Remove from |layers| all layers that are not {{RTCRtpEncodingParameters/active}}, or whose
+        corresponding {{RTCRtpSender}} track has ended.
+    1. If |rid| is not undefined, remove from |layers| all layers whose rid is not |rid|.
```
```suggestion
    1. If |rid| is not undefined, remove from |layers| all layers whose RID is not |rid|.
```

youennf commented on index.bs:
```diff
@@ -1003,32 +1004,35 @@ The <dfn abstract-op>generate key frame algorithm</dfn>, given |promise|, |encod
-    1. If |frame| was generated by a video encoder identified by |task|.`[[rid]]`, run the following steps:
+    1. If |frame| was generated for a layer
+        [=list/contained=] in |task|.`[[layers]]`, remove that layer from |task|.`[[layers]]`.
+        If this causes |task|.`[[layers]]` to become empty, then run the following steps:
```
Let's resolve the promise as soon as we have a frame, that way, whether there are multiple key frames or just one, it is consistent.
